test/test.googleauth.ts

```diff
@@ -1104,7 +1104,7 @@ describe('googleauth', () => {
      async () => {
        assert.notEqual(true, auth.isGCE);
        const scope = nockNotGCE();
-       const isGCE = await auth._checkIsGCE();
+       await auth._checkIsGCE();
```

Comment:
kind of a weird API method; you call `auth._checkIsGCE()` and as a side effect it populates `auth.isGCE`? I'd potentially instead memoize this, and always `await auth._checkIsGCE()`, but on future calls you'll just get a promise that resolves immediately.

----

this has nothing to do with your work :smile:

Reply:
I've been chipping at this thing for a while, but it's still gross 🤣